awc/src/client/connector.rs

```diff
@@ -80,22 +82,62 @@ impl Connector<()> {
         Connector {
             connector: TcpConnector::new(resolver::resolver()).service(),
             config: ConnectorConfig::default(),
-            tls: Self::build_ssl(vec![b"h2".to_vec(), b"http/1.1".to_vec()]),
+            tls: Self::build_tls(vec![b"h2".to_vec(), b"http/1.1".to_vec()]),
         }
     }
 
     /// Provides an empty TLS connector when no TLS feature is enabled.
-    #[cfg(not(any(feature = "openssl", feature = "rustls")))]
-    fn build_ssl(_: Vec<Vec<u8>>) -> OurTlsConnector {
+    #[cfg(not(any(feature = "openssl", feature = "rustls-0_20", feature = "rustls-0_21")))]
+    fn build_tls(_: Vec<Vec<u8>>) -> OurTlsConnector {
         OurTlsConnector::None
     }
 
-    /// Build TLS connector with rustls, based on supplied ALPN protocols
+    /// Build TLS connector with Rustls v0.21, based on supplied ALPN protocols
+    ///
+    /// Note that if other TLS crate features are enabled, Rustls v0.21 will be used.
+    #[cfg(feature = "rustls-0_21")]
+    fn build_tls(protocols: Vec<Vec<u8>>) -> OurTlsConnector {
+        use actix_tls::connect::rustls_0_21::{reexports::ClientConfig, webpki_roots_cert_store};
+
+        let mut config = ClientConfig::builder()
+            .with_safe_defaults()
+            .with_root_certificates(webpki_roots_cert_store())
+            .with_no_client_auth();
+
+        config.alpn_protocols = protocols;
+
+        OurTlsConnector::Rustls021(std::sync::Arc::new(config))
+    }
+
+    /// Build TLS connector with Rustls v0.21, based on supplied ALPN protocols
+    ///
+    /// Note that if other TLS crate features are enabled, Rustls v0.21 will be used.
+    #[cfg(all(
+        all(feature = "rustls-0_20", feature = "openssl"),
+        not(feature = "rustls-0_21"),
+    ))]
```

Comment:
Maybe we can go with something like:
```rust
#[cfg(all(
    any(
        all(feature = "rustls-0_20", feature = "openssl"),
        feature = "rustls-0_20",
    ),
    not(feature = "rustls-0_21"),
))]
```

To avoid needing two functions to rustls 0.20 ?

Reply:
change made in ac95362340c195b60639e48a610e9d7f3677ba18